src/output/render/permissions.rs

```diff
@@ -76,6 +88,33 @@ impl f::Permissions {
     }
 }
 
+impl f::Attributes {
+    pub fn render<C: Colours+FiletypeColours>(&self, colours: &C) -> Vec<ANSIString<'static>> {
+        let bit = |bit, chr: &'static str, style: Style| {
+            if bit { style.paint(chr) }
+              else { colours.dash().paint("-") }
+        };
+
+        vec![
+            bit(self.archive,   "a", colours.normal()),
+            bit(self.readonly,  "r", colours.user_read()),
```

Comment:
I didn't create other color for `readonly`, `hidden` & `system`. It looks fine now. And I don't think people would actually complain the naming of the color, since only a handful people would actually read the code 😄  If someone does fancy dedicated colors on Windows, we can add them in the future.

Reply:
Yeah, I think re-using existing colours for this is fine!